src/Titon/Context/Definition/DefinitionFactory.hh

```diff
@@ -57,7 +61,7 @@ class DefinitionFactory {
             return new CallableDefinition($key, $class, (string) $function, $depository);
         }
 
-        throw new ItemNotDefinableException("Cannot create a definition from " . print_r($concrete, true));
+        throw new ItemNotDefinableException(sprintf('Cannot create a definition for %s', $key));
```

Comment:
Perhaps. I know that `sprintf` takes types into account correctly for %s, %d, etc. Seems like a better approach than interpolation.